bokehjs/src/coffee/models/tickers/log_ticker.coffee

```diff
@@ -31,32 +31,31 @@ export class LogTicker extends AdaptiveTicker
         factors = range(start_factor, end_factor + 1)
 
       ticks = (factor * interval for factor in factors when factor != 0)
+      ticks = ticks.filter((tick) -> data_low <= tick <= data_high)
 
-      if num_minor_ticks > 1
+      if num_minor_ticks > 0 and ticks.length > 0
         minor_interval = interval / num_minor_ticks
-        minor_offsets = (i*minor_interval for i in [1..num_minor_ticks])
-        for x in minor_offsets
+        minor_offsets = (i*minor_interval for i in [0...num_minor_ticks])
+        for x in minor_offsets[1..minor_offsets.length]
           minor_ticks.push(ticks[0]-x)
         for tick in ticks
           for x in minor_offsets
             minor_ticks.push(tick+x)
     else
-      startlog = Math.ceil(log_low)
-      endlog = Math.floor(log_high)
+      startlog = Math.ceil(log_low * 0.9999)
```

Comment:
I'm unsure about the wisdom of dealing with floating point errors in this way. The issue is for a range between 1 and 1000:
```coffeescript
data_low, data_high = 1, 1000
log_log, log_high = 1, 2.999999997
start_log, end_log = 1, 2
```
so we end up missing a right-hand tick. This adds a small fudge factor that's supposed to be larger than the floating point error, so the tick is present.